ecs: update docs for new structure of ability_bonuses response field

ecshreve opened the pull request:
## What does this do?
- documentation updates in relation to this pr: https://github.com/bagelbits/5e-database/pull/309

## How was it tested?
- localhost 

## Is there a Github issue this is resolving?
https://github.com/bagelbits/5e-srd-api/issues/126

## Here's a fun image for your troubles
![random photo - update me](https://picsum.photos/200)

bagelbits commented on src/views/partials/doc-resource-races.ejs:
```diff
@@ -285,6 +288,48 @@
   </tbody>
 </table>
 
+<h3>GET api/races/{index}/ability-bonuses/</h3>
```
This endpoint doesn't exist. Should it?

ecshreve replied:
That was one of my questions I left up in the PR description, along with a few other ones that I'm not sure the answers to, hoping for feedback / input 🙂 

> - where should this type live?
> - should it be in it's own file?
> - should it be under the Race section?
> - should resources of this type be available via their own endpoint?

bagelbits replied:
You've learned my terrible secret. My reading comprehension is crap.

bagelbits replied:
I think for now, it doesn't make sense for this to endpoint to exist. Especially with Graphql so close. However, there is a section on the docs for common use patterns which it makes sense for this to live for now.

ecshreve replied:
I agree, that sounds like the right thing to do. I'll tweak this branch a bit tonight. Thanks!

ecshreve commented:
@bagelbits i added a commit to add `ability-bonus` to the common models section of the docs and removed the reference to the specific endpoint for it. I think this is in line with what we were talking about in the previous comments on this PR?

<img width="646" alt="Screen Shot 2020-10-30 at 12 14 52 AM" src="https://user-images.githubusercontent.com/1425775/97671169-f8406480-1a44-11eb-8f3d-433fe72ea032.png">

bagelbits commented:
@ecshreve I'm hesitant to call the data shape `Ability Bonus`. This shape can easily be generalized to any valued `APIReference`. Though, that might be preemptive generalization.

ecshreve commented:
> @ecshreve I'm hesitant to call the data shape `Ability Bonus`. This shape can easily be generalized to any valued `APIReference`. Though, that might be preemptive generalization.

I see what you're saying, we could call it `APIReferenceWithValue` (or something along those lines)? I waffled a bit thinking about this earlier 🤔

ecshreve commented:
I'm leaning towards generalizing now (even if it feels preemptive) because it's usually more difficult and sometimes annoying to generalize later.

ecshreve commented:
As far as naming goes I'm definitely open to input, some things that came to mind were `Bonus` or `Modifier`, each representing an `APIReference` and an `integer`

fergcb commented:
I would be a bit hesitant to generalise this structure too much. Of course, there are plenty of instances where an object has an APIReference and one other value, but often the purposes of those values don't overlap. E.g. compare ability bonuses to starting equipments' quantitied items.

It would be a shame to ditch the inherent semantics in the attribute names for the sake of generalisation, I feel.

ecshreve commented:
> I would be a bit hesitant to generalise this structure too much. Of course, there are plenty of instances where an object has an APIReference and one other value, but often the purposes of those values don't overlap. E.g. compare ability bonuses to starting equipments' quantitied items.
> 
> It would be a shame to ditch the inherent semantics in the attribute names for the sake of generalisation, I feel.

This also makes a lot of sense. I'm going to think on this a bit more 🤔 (I appreciate the input, and am very open to more suggestions)

ecshreve commented:
One option would be to drop the shape definition for `Ability Bonus` somewhere (either top or bottom) under the `Race` section. But from what I can tell, the only data shape definitions that exist without their own dedicated endpoint are the ones that live under the `Common Models` section. And I'm always hesitant to introduce a variation like that for something that would only be used once. 🤔

fergcb commented:
Personally, I think `Ability Bonus` is absolutely fine to stay in `Common Models`.
